plugin/evm/limitorders/memory_database.go

```diff
@@ -228,31 +228,37 @@ func (db *InMemoryDatabase) UpdateNextFundingTime(nextFundingTime uint64) {
 	db.NextFundingTime = nextFundingTime
 }
 
-func (db *InMemoryDatabase) GetLongOrders(market Market) []LimitOrder {
+func (db *InMemoryDatabase) GetFulfillableOrders(market Market, lastBlockTime uint64) ([]LimitOrder, []LimitOrder) {
+	db.mu.Lock()
+	defer db.mu.Unlock()
+
 	var longOrders []LimitOrder
-	for _, order := range db.OrderMap {
+	var shortOrders []LimitOrder
+	for orderId, order := range db.OrderMap {
+		if order.Expiry.Uint64() <= lastBlockTime {
+			deleteOrder(db, orderId)
```

Comment:
Since expiry only happens based on time and doesn't depend on block number, I think we could delete it once we're sure that the time has passed(irrespective of accepted block number).

However, if we still want to use SetOrderStatus to make it consistent with other status updates, then which block number should be used? Order's original block number or current latest accepted block number?

Reply:
I'm setting the current block number(block being produced) while marking it expired now. Also removed `lastBlockTime` and using `time.Now()` because
- since we're using current block number, we should consider current time instead of head block's time
- there are cases when block is being produced after some time because of inactivity. In that case, using last block's time is incorrect.

Reply:
Using `lastBlockTime` now, as discussed